### ISSUE TITLE
Phase 6.2: Train Entailment Classifier (~10M params)

### ISSUE THREAD

**Issue by marctjones** [marctjones/klareco#267]
## Parent Issue
Part of Epic #252: Enhanced AST-Aware Retriever Implementation

## Overview
Train a classifier to determine if a document entails/supports an answer to a query.

## Model Architecture
```
Input: (query_ast, doc_ast) → Joint encoding → 3-class output

Classes:
- ENTAILS: Document supports answering the query
- NEUTRAL: Document neither supports nor contradicts
- CONTRADICTS: Document contradicts expected answer

Architecture:
- Shared encoder (from semantic similarity model)
- Cross-attention layer
- Classification head (3 classes)
```

## Training Data
Source: Q&A benchmark + synthetic generation

### Entailment examples
- Query: "Kiu fondis Esperanton?"
- Doc: "Zamenhof kreis la internacian lingvon en 1887."
- Label: ENTAILS

### Neutral examples
- Query: "Kiu fondis Esperanton?"
- Doc: "Esperanto havas 16 gramatikajn regulojn."
- Label: NEUTRAL

### Contradiction examples
- Query: "Kiu fondis Esperanton?"
- Doc (hypothetical): "Neniu fondis Esperanton."
- Label: CONTRADICTS

## Synthetic Data Generation
```python
def generate_contradictions(positive_example):
    """Generate contradiction by negation."""
    # Add "ne" to verb
    # Replace subject with "neniu"
    # Replace positive adjectives with negative
```

## Training Configuration
```yaml
model:
  base_encoder: semantic_similarity  # Transfer learning
  cross_attention_heads: 4
  classifier_hidden: 256
  num_classes: 3

training:
  batch_size: 32
  learning_rate: 5e-5
  epochs: 30
  class_weights: [1.0, 0.5, 2.0]  # Boost contradiction detection
```

## Success Criteria
- [ ] 3-class accuracy >80%
- [ ] Contradiction F1 >70%
- [ ] Model size ~10M parameters
- [ ] Saved to models/entailment_classifier/

**Comment by marctjones** [comment]
Future work (M4 scope). Focus on corpus quality first.
